libs/langchain-ollama/src/tests/chat_models_think.int.test.ts

```diff
@@ -52,3 +52,175 @@ test("test deep seek model with think=true (default)", async () => {
   expect(responseContent).toMatch(/photosynthesis/i); // Check it includes the topic
   expect(responseContent.length).toBeGreaterThan(1);
 });
+
+test("test type safety for thinking parameter values", async () => {
+  // Test that TypeScript accepts all valid string and boolean values
+  const ollamaHigh = new ChatOllama({
+    model: "deepseek-r1:32b",
+    think: "high" as const,
+    maxRetries: 1,
+  });
+  
+  const ollamaMedium = new ChatOllama({
+    model: "deepseek-r1:32b", 
+    think: "medium" as const,
+    maxRetries: 1,
+  });
+  
+  const ollamaLow = new ChatOllama({
+    model: "deepseek-r1:32b",
+    think: "low" as const,
+    maxRetries: 1,
+  });
+  
+  const ollamaTrue = new ChatOllama({
+    model: "deepseek-r1:32b",
+    think: true,
+    maxRetries: 1,
+  });
+  
+  const ollamaFalse = new ChatOllama({
+    model: "deepseek-r1:32b",
+    think: false,
+    maxRetries: 1,
+  });
+
+  // All should be properly instantiated
+  expect(ollamaHigh).toBeDefined();
+  expect(ollamaMedium).toBeDefined();
+  expect(ollamaLow).toBeDefined();
+  expect(ollamaTrue).toBeDefined();
+  expect(ollamaFalse).toBeDefined();
+  
+  // Quick test that string values work in practice
+  const res = await ollamaHigh.invoke([
+    new HumanMessage({ content: "How many r in the word strawberry?" })
+  ]);
+  
+  expect(res).toBeDefined();
+  expect(typeof res.content).toBe("string");
+  expect(res.content.length).toBeGreaterThan(0);
```

Comment:
No reason to use the model if we just do type checks. In `v1` branch we have migrated to Vitest which allows to create type tests.